src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    internal static class HostingLoggerExtensions
+    {
+        public static IDisposable RequestScope(this ILogger logger, HttpContext httpContext)
+        {
+            return logger.BeginScopeImpl(new HostingRequestScope(httpContext));
+        }
+
+        public static void RequestStarting(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(LogLevel.Information, 1, new HostingRequestStarting(httpContext), null, HostingRequestStarting.Callback);
+            }
+        }
+
+        public static void RequestFinished(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(LogLevel.Information, 2, new HostingRequestFinished(httpContext), null, HostingRequestFinished.Callback);
+            }
+        }
+
+        private class HostingRequestScope : ILogValues
+        {
+            private readonly HttpContext _httpContext;
+
+            private FeatureReference<IHttpRequestIdentifierFeature> _requestIdentifierFeatureReference;
```

Comment:
Will merge back into other PR, and add comment changes - hopefully it will flow though soon :)